test/bedrock_Azure_mc_test.go

```diff
@@ -0,0 +1,234 @@
+	addIPtoYAML(configFile, string(centralIP))	
+	k8s.KubectlApply(t, options3, configFile)
+
+	//Test Case 4: Validate Traffic Manager
+	testTM_URL := "http://" + tm_dnsprefix + ".trafficmanager.net"
+
+	// It can take several minutes or so for the app to be deployed, so retry a few times
+	maxRetries := 60
+	timeBetweenRetries := 5 * time.Second
+
+	//Verify that we get a 200 OK response and response text contains `Cats` otherwise clean up AKS load balancer and destroy resources
+	//Bedrock is using the azure-vote.yaml service that provisions a stateless simple voting app using redis on all clusters
+	_reqErr := http_helper.HttpGetWithRetryWithCustomValidationE(t, testTM_URL, maxRetries, timeBetweenRetries, func(status int, body string) bool {
+		return status == 200 && strings.Contains(body, `"Cats"`)
+	})
+
+	if _reqErr != nil {
+		fmt.Println("Error validating Traffic Manager - Removing cluster load balancer and Destroying resources")
+		_clean, cleanErr := k8s.RunKubectlAndGetOutputE(t, options, "delete", "service", "azure-vote-front")
+		if cleanErr != nil || !strings.Contains(_clean, "delete") {
+			t.Fatal(cleanErr)
+		} else {
+			fmt.Println("Clean verification for West Cluster complete")
+		}
+		_clean2, cleanErr2 := k8s.RunKubectlAndGetOutputE(t, options2, "delete", "service", "azure-vote-front")
+		if cleanErr2 != nil || !strings.Contains(_clean2, "delete") {
+			t.Fatal(cleanErr2)
+		} else {
+			fmt.Println("Clean verification for East Cluster complete")
+		}
+		_clean3, cleanErr3 := k8s.RunKubectlAndGetOutputE(t, options3, "delete", "service", "azure-vote-front")
+		if cleanErr3 != nil || !strings.Contains(_clean3, "delete") {
+			t.Fatal(cleanErr3)
+		} else {
+			fmt.Println("Clean verification for Central Cluster complete")
+		}
+		//Sleep job for 2 minutes while load balancer deallocates
+		time.Sleep(120 * time.Second)
+		t.Fatal(cleanErr)
+	} else {
+		fmt.Println("Traffic Manager Validation successful")
+	}
+	//Clean up Cluster load balancers
+	fmt.Println("Removing cluster load balancer and Destroying resources")
+	_clean, cleanErr := k8s.RunKubectlAndGetOutputE(t, options, "delete", "service", "azure-vote-front")
+	if cleanErr != nil || !strings.Contains(_clean, "delete") {
+		t.Fatal(cleanErr)
+	} else {
+		fmt.Println("Clean verification for West Cluster complete")
+	}
+	_clean2, cleanErr2 := k8s.RunKubectlAndGetOutputE(t, options2, "delete", "service", "azure-vote-front")
+	if cleanErr2 != nil || !strings.Contains(_clean2, "delete") {
+		t.Fatal(cleanErr2)
+	} else {
+		fmt.Println("Clean verification for East Cluster complete")
+	}
+	_clean3, cleanErr3 := k8s.RunKubectlAndGetOutputE(t, options3, "delete", "service", "azure-vote-front")
+	if cleanErr3 != nil || !strings.Contains(_clean3, "delete") {
+		t.Fatal(cleanErr3)
+	} else {
+		fmt.Println("Clean verification for Central Cluster complete")
+	}
+	//Sleep job for 2 minutes while load balancer deallocates
+	time.Sleep(120 * time.Second)
```

Comment:
Do we need to wait 2 minutes for everything to deallocate?  (Can we just let Azure do this in the background to make CI/CD faster?)

Reply:
Once the validation script completes, terraform begins to destroy resources including the `public_ip`. This throws an error half of the time because the load-balancer has not been fully de-allocated within the aks cluster. Currently waiting for 2 minutes to make sure load-balancer is removed to avoid the error.

Reply:
Ok - looks good to me then.